docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,340 @@ To make the actions more user-friendly you can also use alternating buttons as
+input got within a certain range of the limit. In order to do that, we'll need to be
+sure that the `TextLengthChecker` customisation is loaded ahead of the `CharacterCounter` 
+customisation. 
+
+First let's update the character counter to show characters _remaining_, which is
+much more useful. We'll also update the API to allow a `warningBuffer` prop. This is
+the amount of characters the input can be within the `limit` before the warning shows.
+
+__my-module/js/components/CharacterCounter.js__
+```js
+import React from 'react';
+
+const CharacterCounter = (TextField) => (props) => {
+    const { warningBuffer, limit, value: { length } } = props;
+    const remainingChars = limit - length;
+    const showWarning = length + warningBuffer >= limit;
+    return (
+        <div>
+            <TextField {...props} />
+            {showWarning &&
+            	<small>Characters remaining: {remainingChars}</small>
+            }
+        </div>
+    );
+}
+
+export default CharacterCounter;
+```
+
+Now, when we apply this customisation, we need to be sure it loads _after_ the length
+checker in the middleware chain, as it relies on the prop `limit`. We can do that by specifying priority using `before` and `after`
+metadata to the customisation.
+
+__my-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import CharacterCounter from './components/CharacterCounter';
+import TextLengthChecker from './components/TextLengthChecker';
+Injector.update(
+	{
+		name: 'my-module',
+		after: 'my-other-module',
+	},
+	wrap => {
+		wrap('TextField', CharacterCounter);
+	}
+);
+Injector.update(
+	{
+		name: 'my-other-module',
+	    before: 'my-module',
+	},
+	wrap => {
+		wrap('TextField', TextLengthChecker);
+	}
+);
+```
+
+Now, both components play together nicely.
+
+![](../../../_images/react-di-3.png)
+
+### Registering new React components
+
+If you've created a module using React, it's a good idea to afford other developers an 
+API to enhance those components. To do that, simply register them with `Injector`.
+
+__my-public-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+
+Injector.register('MyComponent', MyComponent);
+```
+
+Now other developers can customise your components with `Injector.update()`.
+
+Note: Overwriting components by calling `register()` multiple times for the same
+service name is discouraged, and will throw an error. Should you really need to do this,
+you can pass `{ force: true }` as the third argument to the `register()` function.
+
+### Using the injector within your component
+
+If your component has dependencies, you can add the injector via context using the `withInjector`
+higher order component.
+
+__my-module/js/components/Gallery.js__
+```js
+import React from 'react';
+import { withInjector } from 'lib/Injector';
+
+class Gallery extends React.Component {
+  render() {
+    const GalleryItem = this.context.injector.get('GalleryItem');
```

Comment:
It might be useful to show this line earlier on? Up until this point, any React dev will scratch their head on how all these customisations are actually applied.